app/src/main/java/edu/uwstout/p2pchat/NickNameFragment.java

```diff
@@ -82,66 +83,72 @@ public View onCreateView(final LayoutInflater inflater, final ViewGroup containe
         //Inilizes listview for nicknames here.
         ListView listView = (ListView) view.findViewById(R.id.nameListView);
 
+
+        //Get View model.
         liveData = getViewModel(getActivity().getApplication()).getPeers();
-        liveData.observeForever(new Observer<List<Peer>>() {
-            @Override
-            public void onChanged(List<Peer> peers) {
-                listView.setAdapter(getNickNameAdapter(peers));
-            }
-        });
+
+        liveData.observeForever(new Observer<List<Peer>>()
+                                {
+                                    @Override
+                                    public void onChanged(List<Peer> peers)
+                                    {
+                                        ArrayAdapter adapter = getNickNameAdapter(peers);
+                                        listView.setAdapter(adapter);
+                                        adapter.notifyDataSetChanged();
+
+                                    }
+                                });
+
```

Comment:
Not an issue, but this seems like a redundant amount of indentation. Did you run the code cleanup?